### PR TITLE
Granularity mnrva 808

### PR DESCRIPTION
**Pull request by FarajiA**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-808

 ## Description:
 Added function to add query params to our metrics api requests depending on the date span. Rules created:  

- Rule 1: when under 48hrs raw data is returned no need for aggregator or granularity
- Rule 2: when start time over 48hrs add aggregator other than raw for example min, max, sum, avg (use avg )

Also removed isValidDate utility function for moment.js library validation already present 

 ## Testing:
 - create date spanning more than 48hrs, the network request will include the necessary query params

 ## Screenshots:
(if applicable)

### CODE REVIEW

**Review by pras7430 — APPROVED** [review]
seems me fine!!

File: src/app/_features/visualize/components/timerange/timerange.component.ts [hunk 1]
```diff
@@ -15,62 +23,50 @@ import { isValidDate } from '@minerva/_shared/utils';
...
-    this.date.start=null;
+    this.date.start = null;
     this.date.end = null;
   }
 
 
   timeRangeUpdate() {
     let start = new Date((this.date.start).setSeconds(0)).toISOString();
     let end = new Date((this.date.end).setSeconds(0)).toISOString();
-    this.date.duration= null;
```
Comment (lovedeep92) [inline src/app/_features/visualize/components/timerange/timerange.component.ts:73]: Nice .. that is what I confused about as it is not used anywhere!

File: src/app/_services/metrics/metrics.service.spec.ts [hunk 2]
```diff
@@ -80,7 +81,7 @@ describe('MetricsService', () => {
   it('should compose query params for requests', () => {
     service.start = 'xiisusiis';
     expect(service['queryParams']()).toEqual({
-      start: 'xiisusiis-ago'
+      start: 'xiisusiis'
```
Comment (lovedeep92) [inline src/app/_services/metrics/metrics.service.spec.ts:84]: what is this value means?

File: src/app/_services/metrics/metrics.service.ts [hunk 3]
```diff
@@ -99,14 +100,65 @@ export class MetricsService {
...
+    let duration = presetDates.find(({ value }) => value === start)?.value;
+    if (duration === presetDates[0].value ||duration === presetDates[1].value ||
+      duration === presetDates[2].value ) {
+        return {
+          start
+        };
+    }
+
+    // if start is in preset array & over 2 days set pertaining params
+    if (duration === presetDates[3].value || duration === presetDates[4].value) {
```
Comment (lovedeep92) [inline src/app/_services/metrics/metrics.service.ts:123]: can't we do it with looping?
and check for value and see if it is more than 2 days then add granularity ?

File: src/app/_services/metrics/metrics.service.spec.ts (hunk 2 above)
Comment (FarajiA) [reply src/app/_services/metrics/metrics.service.spec.ts:84]: It means nothing. It's just rubbish

File: src/app/_services/metrics/metrics.service.ts (hunk 3 above)
Comment (FarajiA) [reply src/app/_services/metrics/metrics.service.ts:123]: Show me

File: src/app/_services/metrics/metrics.service.spec.ts (hunk 2 above)
Comment (lovedeep92) [reply src/app/_services/metrics/metrics.service.spec.ts:84]: 😆

File: src/app/_services/metrics/metrics.service.ts (hunk 3 above)
Comment (lovedeep92) [reply src/app/_services/metrics/metrics.service.ts:123]: Something like :-

```
presetDates.forEach((key, value) => {
    if(key === 'WEEK' || key === 'MONTH') {
         granularity = 'pt1h';
            aggregator = 'avg';
    } else {
         return {
          start
        };
    }
})
```

Comment (FarajiA) [reply src/app/_services/metrics/metrics.service.ts:115]: Hey the main problem with this route is it looks to not exit the calling function but does exit the `foreach` loop. Feel free to attempt.

**Pull request merged**